From 0db44b74a2565e2092324e0450b891e5a7ccf2fb Mon Sep 17 00:00:00 2001
From: Oliver Heger <oheger@apache.org>
Date: Tue, 29 Apr 2014 20:35:48 +0000
Subject: [PATCH] [CONFIGURATION-573] Added a test for namespaces.

The test checks whether XPathExpressionEngine now handles expressions
referencing nodes and attributes with namespaces correctly.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/configuration/trunk@1591092 13f79535-47bb-0310-9956-ffa450edef68
---
 .../TestXPathExpressionEngineInConfig.java    | 33 +++++++++++++++++--
 1 file changed, 30 insertions(+), 3 deletions(-)

diff --git a/src/test/java/org/apache/commons/configuration/tree/xpath/TestXPathExpressionEngineInConfig.java b/src/test/java/org/apache/commons/configuration/tree/xpath/TestXPathExpressionEngineInConfig.java
index 2070585c5a..f493d52dd5 100644
--- a/src/test/java/org/apache/commons/configuration/tree/xpath/TestXPathExpressionEngineInConfig.java
+++ b/src/test/java/org/apache/commons/configuration/tree/xpath/TestXPathExpressionEngineInConfig.java
@@ -17,8 +17,14 @@
 package org.apache.commons.configuration.tree.xpath;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.StringReader;
+import java.util.Iterator;
 
 import org.apache.commons.configuration.XMLConfiguration;
+import org.apache.commons.configuration.ex.ConfigurationException;
+import org.apache.commons.configuration.io.FileHandler;
 import org.junit.Before;
 import org.junit.Test;
 
@@ -26,9 +32,6 @@
  * A test class for XPathExpressionEngine that tests the engine integrated into
  * a hierarchical configuration.
  *
- * @author <a
- *         href="http://commons.apache.org/configuration/team-list.html">Commons
- *         Configuration team</a>
  * @version $Id$
  */
 public class TestXPathExpressionEngineInConfig
@@ -115,4 +118,28 @@ public void testAddPropertyComplexStructures()
         assertEquals("Wrong field type", "int",
                 config.getString("tables/table[1]/fields/field[1]/@type"));
     }
+
+    /**
+     * Tests whether configuration properties with a namespace can be handled.
+     */
+    @Test
+    public void testPropertiesWithNamespace() throws ConfigurationException
+    {
+        String xml =
+                "<Config>\n"
+                        + "<dsig:Transforms xmlns:dsig=\"http://www.w3.org/2000/09/xmldsig#\">\n"
+                        + "  <dsig:Transform Algorithm=\"http://www.w3.org/TR/1999/REC-xpath-19991116\">\n"
+                        + "    <dsig:XPath xmlns:ietf=\"http://www.ietf.org\" xmlns:pl=\"http://test.test\">self::pl:policy1</dsig:XPath>\n"
+                        + "  </dsig:Transform>\n"
+                        + "  <dsig:Transform Algorithm=\"http://www.w3.org/TR/2001/REC-xml-c14n-20010315\"/>\n"
+                        + "</dsig:Transforms>" + "</Config>";
+        FileHandler handler = new FileHandler(config);
+        handler.load(new StringReader(xml));
+
+        for (Iterator<String> it = config.getKeys(); it.hasNext();)
+        {
+            String key = it.next();
+            assertNotNull("No value for " + key, config.getString(key));
+        }
+    }
 }
